repositories/db_executor_getter.go

```diff
@@ -20,10 +21,21 @@ type databaseSchemaGetter interface {
 }
 
 type Executor interface {
-	transactionOrPool
+	TransactionOrPool
 	databaseSchemaGetter
 }
 
+type Transaction interface {
+	databaseSchemaGetter
+	Exec(ctx context.Context, sql string, arguments ...any) (pgconn.CommandTag, error)
+	Query(ctx context.Context, sql string, args ...any) (pgx.Rows, error)
+	QueryRow(ctx context.Context, sql string, args ...any) pgx.Row
+	// Begin(ctx context.Context) (pgx.Tx, error)
+	Commit(ctx context.Context) error
+	Rollback(ctx context.Context) error
+	RawTx() pgx.Tx
```

Comment:
Adding this was really the main initial objective of this PR. However, to be able to do this I needed to distinguish between just passing a generic pool/tx, or specifically passing a transaction (and the check should be at compile time, I don't want runtime errors "oh I was expecting a transaction").
It just made sense to take the time and rework it so we have finer control on what usecase/repo methods that _receive_ an executor, actually expect a transaction.